### DISCUSSION TITLE
Extending spec for common extensions

### DISCUSSION THREAD

**Discussion by yairlenga** [handlebars-lang/handlebars.js#1823] [Ideas]
Hi,

My team is using multiple implementations of handlebars - us in the browser, Java on the server and python for Backend. The systems share templates (templates are developed interactively in a briwser, and executed in the backend).

one challenge that e have is that the “spec” is very minimal, too minimal. We found that we need standard operation like: eq (equal), is_number (validation), match (regular expression). The problem is that each extension language provides slightly different variants for those operator, with different names.

While I understand (and appreciate) keeping the language small, I think it will help if the “spec” can be extended with SMALL number of standard operator (with standard naming). Implementations of handlebar will deliver standard implementation, but will be able to provide enhanced versions with different operator name. Users will know for sure what is the behavior.

my small list:
* eq (equal operator, by value, strict typing, works on values, array, or objects, no casting)
* Is_number, Is_integer, Is_string, 
* Match (regex match, for verification) - true/false
* number formatting - printf or similar - needed for reporting - decimal, thousands separator, sign, etc.

any idea/suggestions on how to make this happen.I think the starting point is a small spec, but not sure what is the process.

**Comment by jaylinski** [comment]
I think this should be an issue in https://github.com/handlebars-lang/handlebars-parser, since it contains the definition for the Handlebars language AST.
